plasma_framework/test/src/exits/payment/PaymentProcessInFlightExit.test.js

```diff
@@ -429,17 +404,46 @@ contract('PaymentInFlightExitRouter', ([_, ifeBondOwner, inputOwner1, inputOwner
             });
         });
 
-        describe('When the exit is non canonical, and inputs are piggybacked', () => {
+        describe('When any input is spent, given the challenge game result is canonical', () => {
+            beforeEach(async () => {
+                const exit = await buildInFlightExitData();
+                this.dummyExitId = 666;
+                exit.isCanonical = true;
+                await this.exitGame.setInFlightExit(this.dummyExitId, exit);
+
+                // flags the first input and piggyback the two ETH inputs (first and second)
+                await this.exitGame.proxyFlagOutputSpent(TEST_OUTPUT_ID_FOR_INPUT_1);
+                await this.exitGame.setInFlightExitInputPiggybacked(this.dummyExitId, 0);
+                await this.exitGame.setInFlightExitInputPiggybacked(this.dummyExitId, 1);
+            });
+
+            it('should be treat as non canonical', async () => {
+                const { logs } = await this.exitGame.processExit(this.dummyExitId, VAULT_ID.ETH, ETH);
+                const inputIndexForInput2 = 1;
+                await expectEvent.inLogs(
```

Comment:
It would be good, if we could check that there was no event for inputIndex 0.

Reply:
added the test under non-canonical test cases (describe) instead of here.